Source/KSCrash/Source/KSCrash/Recording/BSG_KSCrash.m

```diff
@@ -306,41 +316,66 @@ - (void)deleteAllReports {
     [self.crashReportStore deleteAllFiles];
 }
 
+- (NSArray<BugsnagThread *> *)captureThreads:(NSException *)exc depth:(int)depth {
+    NSArray *addresses = [exc callStackReturnAddresses];
+    int numFrames = (int) [addresses count];
+    uintptr_t *callstack;
+
+    if (numFrames > 0) {
+        depth = 0; // reset depth if the stack does not need to be generated
+        callstack = malloc(numFrames * sizeof(*callstack));
+
+        for (NSUInteger i = 0; i < numFrames; i++) {
+            callstack[i] = [addresses[i] unsignedLongValue];
+        }
+    } else {
+        // generate a backtrace. This is required for NSError for example,
+        // which does not have a useful stacktrace generated.
+        numFrames = 100;
+        callstack = malloc(numFrames * sizeof(*callstack));
+
+        BSG_KSLOG_DEBUG("Fetching call stack.");
+        numFrames = backtrace((void **)callstack, numFrames);
+        if (numFrames <= 0) {
+            BSG_KSLOG_ERROR(@"backtrace() returned call stack length of %d", numFrames);
+            numFrames = 0;
+        }
+    }
+
+    char *trace = bsg_kscrash_captureThreadTrace(depth, numFrames, callstack);
```

Comment:
This method currently returns a global variable which can be mutated on another thread once `bsg_kscrash_captureThreadTrace` has resumed thread execution.